test/remote/remote_poli_pay_test.rb

```diff
@@ -37,12 +48,32 @@ def test_bad_password
                  error.message
   end
 
-  def test_required_fields
-    options = transaction_options.except(:homepage_url)
-    helper = PoliPay::Helper.new('22TEST', @login, options)
-    assert_raise KeyError.new('key not found: :homepage_url') do
-      helper.credential_based_url
+  def test_merchant_reference_field
+    reference = '22+TEST'
+    form_fields = raw_url_fields.merge('MerchantReference' => reference)
+    begin
```

Comment:
The problem I face here is I want to inspect the error object, I'm not even sure how you would do this in rspec. The info I want isn't in the "message" field which is easy to test.

I'm finding this is a pattern that does re-occur, so this could be abstracted into a helper perhaps like:
assert_raise_with_custom_message Exception, method, value

e.g.
assert_raise_with_custom_message PoliPay::UrlInterface::UrlRequestError, :error_message, SOME_LONG_STRING